### ISSUE TITLE
IP Address Lockout Bypass

### ISSUE THREAD

**Issue by neonbunny** [dreamsolution/django-auth-policy#10]
The code which determines the source IP address of a login uses the following logic:

```
remote_addr = (self.request.META.get('HTTP_X_REAL_IP') or self.request.META.get('REMOTE_ADDR'))
```

This means an attacker can override their source IP address by supplying an X-Real-IP header containing an IP address different than their own (assuming the attacker's header isn't replaced/stripped by an intermediate system). A brute force attack is therefore still possible by cycling through values in this header.

This library should only support a single mechanism of determining the visitor's IP address which is configured by the library user to match the deployment environment (e.g. direct Internet connect or behind a load balancer)

**Comment by rudolphfroger** [comment]
Thanks for you feedback. Do you think it would be enough to add a configuration setting to determine the HTTP header to use?

**Comment by neonbunny** [comment]
Yes - that would be my suggestion. The default setting when a header isn't configured should be REMOTE_ADDR - but if a header is configured, then ONLY that one header should be used.

**Comment by rudolphfroger** [comment]
Fixed in commit df86442
